File: 1 - BEGINNER/Java/1040 - Média 3.java
import java.io.IOException;
import java.util.Scanner;
import java.text.DecimalFormat;

public class Main {
 
    public static void main(String[] args) throws IOException {
 
        Scanner input = new Scanner(System.in);
        DecimalFormat df = new DecimalFormat("0.0");
        
        double n1 = input.nextDouble();
        double n2 = input.nextDouble();
        double n3 = input.nextDouble();
        double n4 = input.nextDouble();
        double media = (n1 * 2 + n2 * 3 + n3 * 4 + n4 * 1) / 10;
        
        System.out.print("Media: " + df.format(media) + "\n");
        if(media < 5.0){
            System.out.print("Aluno reprovado.\n");
        } else if(media >= 7.0){
            System.out.print("Aluno aprovado.\n");
        } else if(media >= 5.0 && media < 7.0){
            System.out.print("Aluno em exame.\n");
            double n5 = input.nextDouble();
            System.out.print("Nota do exame: " + df.format(n5) + "\n");
            double mediaNova = (media + n5) / 2;
            if(mediaNova >= 5.0){
                System.out.print("Aluno aprovado.\n");
            } else{
                System.out.print("Aluno reprovado.\n");
            }
            System.out.print("Media final: " + df.format(mediaNova) + "\n");
        }
        
        input.close();
 
    }
 
}

File: 1 - BEGINNER/C/1040 - Média 3.c
#include <stdio.h>
 
int main() {
 
    float n1, n2, n3, n4, media, media_final, exame;
    
    scanf("%f %f %f %f", &n1, &n2, &n3, &n4);
    
    media = (n1 * 2 + n2 * 3 + n3 * 4 + n4 * 1) / 10.0;
    
    printf("Media: %.1f\n", media);
    
    if(media >= 5.0 && media < 7.0){
        
        printf("Aluno em exame.\n");
        scanf("%f", &exame);
        printf("Nota do exame: %.1f\n", exame);
        media_final = (media + exame) / 2.0;
        
        if(media_final >= 5.0){
            printf("Aluno aprovado.\n");
            printf("Media final: %.1f\n", media_final);
        } else{
            printf("Aluno reprovado.\n");
        }
        
    } else if(media <= 5){
        printf("Aluno reprovado.\n");
    } else if(media >= 7){
        printf("Aluno aprovado.\n");
    }
 
    return 0;
}